Feature/create pdf generator service

FAEZ10 opened the pull request:
# Pull Request Templates

## Type of change

-   [x] Feature
-   [ ] Fix

## Description

This pull request introduces the functionality to create and download invoices as PDFs. The main changes include:

- Added functionality to create and download invoices.
- Created a directory 'invoices' at the project root to store generated invoice PDFs.
- Updated `InvoiceController` to handle the generation and downloading of invoice PDFs.
- Integrated Dompdf for PDF generation with necessary configurations.
- Ensured images are correctly displayed in the PDFs using absolute URLs.
- Improved invoice template for better styling and UI in the generated PDFs.

## How Has This Been Tested?

The following tests were performed to verify the changes:

- Verified that invoices can be created and downloaded as PDFs.
- Checked that the PDF contains the correct invoice details and styling.
- Tested image rendering in the PDFs using absolute URLs.

**Test Configuration**:

-   Symfony version : >=6.4
-   PHP version : >=8.3.2

## Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have added tests that prove my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules

FAEZ14 committed:
Merge branch 'dev' into feature/create-PdfGeneratorService

FAEZ14 committed:
Merge branch 'feature/handle-invoice-number' into feature/create-PdfGeneratorService

FAEZ14 committed:
feat: implement creation and download functionality for invoices

johnnyhelloworld commented on src/Service/PdfGeneratorService.php:
```diff
@@ -0,0 +1,35 @@
+
+        $html = $this->twig->render($template, $data);
+        $options = new Options();
+        // $options->set('isRemoteEnabled', true);
```
remove dead code

DaniOmer replied:
Sometimes dead code can be helpful for debugging

DaniOmer committed:
Merge remote-tracking branch 'origin/dev' into feature/create-PdfGeneratorService

DaniOmer committed:
Remove duplicated import from Invoice Service.

FAEZ14 committed:
Merge branch 'dev' into feature/create-PdfGeneratorService

FAEZ14 committed:
Fix : change uid in invoice_number for display

FAEZ14 committed:
Merge remote-tracking branch 'origin/feature/create-PdfGeneratorService' into feature/create-PdfGeneratorService

DaniOmer committed:
Adding a showPdf method to PDF generator service to enable user to preview invoice and quotation as PDF.

DaniOmer committed:
Replacing the generatePdf by downloadPdf method to avoid downloading the file into the project directory on each user download

DaniOmer committed:
Implementing a method to generate pdf binary content

DaniOmer committed:
Implementing the logic to send the generate invoice to user by email after accepting the related quotation:
- Creating a method to send invoice by email
- Adding some validation to validateQuotation route

DaniOmer committed:
Implementing a template for quotation pdf generation

DaniOmer committed:
Update SendQuotationMail method by replacing csv file by pdf
